src/components/TransactionsTracker/useTransactionsTracker.ts

```diff
@@ -0,0 +1,43 @@
+import { useEffect, useRef } from 'react';
+import { getTransactionsByHashes as defaultGetTxByHash } from 'apiCalls/transactions';
+import { GetTransactionsByHashesType } from 'contexts/types';
+import { useCheckTransactionStatus } from 'hooks';
+import { useGetPendingTransactions } from 'services/transactions';
+import { getIsTransactionPending } from 'utils';
+
+export interface TransactionsTrackerType {
+  getTransactionsByHash?: GetTransactionsByHashesType;
+}
+
+export function useTransactionsTracker(props?: TransactionsTrackerType) {
+  const { pendingTransactionsArray } = useGetPendingTransactions();
+
+  const getTransactionsByHash =
+    props?.getTransactionsByHash ?? defaultGetTxByHash;
+
+  const checkTransactionStatus = useCheckTransactionStatus();
+  const intervalRef = useRef<ReturnType<typeof setInterval>>();
+
+  const pendingBatches = pendingTransactionsArray.filter(
+    ([sessionId, transactionBatch]) => {
+      const isPending =
+        sessionId != null && getIsTransactionPending(transactionBatch.status);
+      return isPending;
+    }
+  );
+
+  useEffect(() => {
+    if (pendingBatches.length > 0) {
+      intervalRef.current = setInterval(() => {
```

Comment:
This should be used only if the behaviour is not overwritten, otherwise we'll end up with an interval, regardless of the way we check for tx status